build-data-capturing-gradle-samples/capture-thermal-throttling/gradle-thermal-throttling.gradle

```diff
@@ -0,0 +1,139 @@
+import org.gradle.api.services.BuildService
+import org.gradle.api.services.BuildServiceParameters
+import org.gradle.internal.os.OperatingSystem
+
+import java.nio.charset.Charset
+import java.util.concurrent.ConcurrentLinkedQueue
+import java.util.concurrent.Executors
+import java.util.concurrent.TimeUnit
+import java.util.function.Function
+
+/**
+ * This Gradle script captures the thermal throttling level and adds it as a tag.
+ * Some parameters can be tweaked according to your build:
+ * - SAMPLING_INTERVAL_IN_SECONDS, frequency at which the capture command is run
+ * - THROTLLING_LEVEL, list of throttling levels and value ranges (to be compared with captured throttling average value)
+ *
+ * WARNINGS:
+ * - This is supported on MacOS only.
+ *
+ * This should be applied to the root project:
+ * <code> apply from: file('gradle-thermal-throttling.gradle') </code>
+ */
+
+def buildScanApi = project.extensions.findByName('buildScan')
+if (!buildScanApi) {
+  return
+}
+
+// Register Thermal throttling service
+def throttlingServiceProvider = gradle.sharedServices.registerIfAbsent("thermalThrottling", ThermalThrottlingService.class) {}
+
+// Start Thermal throttling service
+throttlingServiceProvider.get()
+
+// Process results on build completion
+buildScan {
```

Comment:
let's make this more similar to the Maven groovy code with `buildScan.buildFinished` together on the same line

Reply:
✅